lib/xcodeproj/project/object/build_configuration.rb

```diff
@@ -74,6 +80,52 @@ def sorted_build_settings
           sorted
         end
 
+        # yes, they are case-sensitive.
+        # no, Xcode doesn't do this for other PathList settings nor other
+        # settings ending in SEARCH_PATHS.
+        ARRAY_SETTINGS = %w(
+          ALTERNATE_PERMISSIONS_FILES
+          ARCHS
+          BUILD_VARIANTS
+          EXCLUDED_SOURCE_FILE_NAMES
+          FRAMEWORK_SEARCH_PATHS
+          GCC_PREPROCESSOR_DEFINITIONS
+          GCC_PREPROCESSOR_DEFINITIONS_NOT_USED_IN_PRECOMPS
+          HEADER_SEARCH_PATHS
+          INFOPLIST_PREPROCESSOR_DEFINITIONS
+          LIBRARY_SEARCH_PATHS
+          OTHER_CFLAGS
+          OTHER_CPLUSPLUSFLAGS
+          OTHER_LDFLAGS
+          REZ_SEARCH_PATHS
+          SECTORDER_FLAGS
+          WARNING_CFLAGS
+          WARNING_LDFLAGS
+        ).freeze
+        private_constant :ARRAY_SETTINGS
+
+        def normalize_array_settings(settings)
```

Comment:
The normalisation methods should probably be private because they don't really matter to users of Xcodeproj (as it's handled internally.)